client/lib/media/actions.js

```diff
@@ -189,6 +209,10 @@ function uploadFiles( uploader, files, site ) {
 							data: data.media[ 0 ],
 						} )
 					);
+
+					reduxDispatch( successMediaItemRequest( siteId, transientMedia.ID ) );
```

Comment:
Does this need to be two distinct actions? It feels a bit [like the actions are `setters`](https://redux.js.org/style-guide/style-guide/#model-actions-as-events-not-setters).

Reply:
Yeah, in this case, it's preferred for keeping things simple - `receiveMedia` will handle multiple items, while `successMediaItemRequest` is for the request of a single media item. 

I agree with you these could be combined into a single action, but note that these action creators were there before creating this PR, so I'm just using them wherever it makes sense. 

That being said, we could always come back and refactor here when everything is using Redux.